### PR TITLE
[KernelGen][Nvidia] Add bucketize operator with Triton kernel

### PR DESCRIPTION
**Pull request by Dingxingdi**
## Summary

Adds a Triton kernel implementation for `torch.bucketize` / `bucketize.Tensor`.

This PR includes:
- Triton kernel implementation for `bucketize`
- Registration in FlagGems operator config
- Accuracy tests for default, `right=True`, and `out_int32=True`
- Core benchmark coverage for NVIDIA

## Testing

Correctness:

```bash
CUDA_VISIBLE_DEVICES=0 python -m pytest tests/test_bucketize.py -m bucketize -vs
```

Result:
- PASS
- 54 accuracy cases passed
- Dtype coverage: `torch.float16`, `torch.float32`, `torch.bfloat16`
- Variants covered: default, `right=True`, `out_int32=True`

Benchmark:

```bash
CUDA_VISIBLE_DEVICES=0 python -m pytest benchmark/test_bucketize.py -m bucketize -s --level core
```

Result:
- PASS
- NVIDIA H20 core benchmark: 15 cases
- Arithmetic mean speedup: 0.9889x

## Performance

| Backend | Accuracy Test | Benchmark | Speedup (mean) | Notes |
|---|---:|---:|---:|---|
| NVIDIA H20 | PASS, 54 cases | PASS, 15 core cases | 0.9889x | CUDA 13.2, Docker container `dxd` |
| Tianshu | N/A | N/A | - | No preserved local record |
| Muxi | N/A | N/A | - | No preserved local record |
| Ascend | N/A | N/A | - | No preserved local record |
| Hygon | N/A | N/A | - | No preserved local record |

## Files Changed

- `src/flag_gems/ops/bucketize.py`: Triton kernel implementation
- `tests/test_bucketize.py`: Accuracy tests
- `benchmark/test_bucketize.py`: Core benchmark
- `src/flag_gems/ops/__init__.py`: Operator export
- `src/flag_gems/__init__.py`: `_FULL_CONFIG` registration
- `conf/operators.yaml`: Operator metadata

### CODE REVIEW

**Review by Dongxu-H — COMMENTED** [review]
Pls fix PR workflow failure

###  Suggestions for Improvement

**1. [`src/flag_gems/ops/bucketize.py:48-51`](https://github.com/flagos-ai/FlagGems/pull/3895/files#diff-xxx) - Hard-coded binary search iterations**

```python
for _ in range(10):  # enough for n_boundaries up to 1024
```

The loop is hard-coded to 10 iterations, which limits `n_boundaries` to 1024. Consider:
- Using `math.ceil(math.log2(max(n_boundaries, 1)))` for dynamic iteration count
- Adding a comment/documenting this limitation in the operator description
- Or adding a runtime check with a clear error message if `n_boundaries > 1024`

---

**2. [`tests/test_bucketize.py:16`](https://github.com/flagos-ai/FlagGems/pull/3895/files#diff-xxx) - Test coverage gaps**

All tests use the same fixed boundaries:
```python
boundaries = torch.tensor([1.0, 3.0, 5.0, 7.0, 9.0], device=flag_gems.device)
```

Consider adding tests for:
- **Integer boundaries**: `torch.tensor([1, 3, 5, 7, 9])` (PyTorch supports this)
- **Empty boundaries**: Already handled in kernel but not tested
- **Single element boundaries**: `torch.tensor([5.0])`
- **Different sizes**: 1, 2, many boundaries
- **Negative boundaries**: `torch.tensor([-5.0, -3.0, 0.0, 3.0])`

---

**3. [`src/flag_gems/ops/bucketize.py:88`](https://github.com/flagos-ai/FlagGems/pull/3895/files#diff-xxx) - Hard-coded BLOCK_SIZE**

```python
BLOCK_SIZE = 1024
```

Consider whether this should be tuned based on input size or made configurable.

---

**Comment by bin913** [comment]
@Dingxingdi Please resolve conflicts

File: benchmark/test_bucketize.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+@pytest.mark.parametrize(
+    "op_name, torch_op, dtypes",
+    [
+        pytest.param(
+            "bucketize",
+            torch.bucketize,
+            consts.FLOAT_DTYPES,
+            marks=getattr(pytest.mark, "bucketize", None),
+        ),
+    ],
```
Comment (tengqm) [inline benchmark/test_bucketize.py]: This is unnecessarily complicating the code.

File: conf/operators.yaml [hunk 2]
```diff
@@ -1219,6 +1219,17 @@ ops:
       - NeuralNetwork
     stages:
       - beta: '5.3'
+  - id: bucketize
+    description: Triton kernel implementation for bucketize.
```
Comment (tengqm) [inline conf/operators.yaml]: The description text is useless

File: conf/operators.yaml [hunk 3]
```diff
@@ -1219,6 +1219,17 @@ ops:
...
+    for:
+      - bucketize.Tensor
+    labels:
+      - aten
+      - KernelGen
+    kind:
+      - Math
+    stages:
+      - alpha: '5.3'
```
Comment (tengqm) [inline conf/operators.yaml]:
```suggestion
      - alpha: '5.4'
```

File: conf/operators.yaml (hunk 2 above)
Comment (Dingxingdi) [reply conf/operators.yaml]: Done. Updated the description to describe the operator semantics

File: benchmark/test_bucketize.py (hunk 1 above)
Comment (Dingxingdi) [reply benchmark/test_bucketize.py]: Fixed. Simplified the benchmark to use GenericBenchmark with a small input_fn and removed the custom Benchmark subclass.

**Review by Caeruleann — APPROVED** [review]
